### PR TITLE
fix: Support deriving ActiveEnum without importing some types

### PR DESCRIPTION
**Pull request by siketyan**
## PR Info

- #1143 (this pull request resolves the issue partially)

- Dependencies:
  - N/A

- Dependents:
  - N/A

## Adds

N/A

## Fixes

- [x] Canonicalised usages of types with `sea_orm::` prefix to enable derives without importing the types
  - **Either importing `sea_orm::sea_query` or depending on `sea_query` crate is still needed due to types derived in `sea_query_derive` crate**

## Breaking Changes

N/A

## Changes

N/A

### CODE REVIEW

File: sea-orm-macros/src/lib.rs [hunk 1]
```diff
@@ -539,6 +539,19 @@ pub fn derive_active_model_behavior(input: TokenStream) -> TokenStream {
 ///         - For `string_value`, value should be passed as string, i.e. `string_value = "A"`
 ///         - For `num_value`, value should be passed as integer, i.e. `num_value = 1` or `num_value = 1i32`
 ///         - Note that only one of it can be specified, and all variants of an enum have to annotate with the same `*_value` macro attribute
+///
+/// # Usage
+///
+/// ```
+/// use sea_orm::{sea_query, EnumIter, DeriveActiveEnum};
```
Comment (siketyan) [inline sea-orm-macros/src/lib.rs:546]: Now we do not need to import Iden, EnumIter nor ActiveEnum, but sea_query is still needed.

**Comment by siketyan** [comment]
Hi @billy1624 , thank you for triaging this pull request!
I think #1151 will NOT be resolved by this changes.

**Review by billy1624 — COMMENTED** [review]
Hey @siketyan, thanks for helping!!

> Hi @billy1624 , thank you for triaging this pull request! I think #1151 will NOT be resolved by this changes.

Yes, you're correct! I'm thing rewriting the derive macros, without using `sea_query::Iden` derive macros.

Would you mind if I merge your PR into another local branch so I can work on top of your work?

I'll create a PR to fix both issues and ask for your help to validate the fix did works. Thanks in advance!! :)

**Comment by siketyan** [comment]
@billy1624 
> Would you mind if I merge your PR into another local branch so I can work on top of your work?

Sure, of course!
So I leave it to you about sea_query stuff.

**Pull request merged**